scripts/genDocs/GenDocs.js

```diff
@@ -0,0 +1,270 @@
+                const { section: sectionName } = currSection
+                const { component: componentName } = currComponent
+                const examplesPath = path.join(filePath.examples, sectionName, componentName)
+
+                return {
+                    ...currComponent,
+                    description: this.getReadme({ sectionName, componentName }),
+                    examples: this.getExampleData({ sectionName, componentName })
+                }
+            })
+            // update component field of current section
+            const updatedSection = { ...currSection, components }
+            // add components for section back to accumulator
+            return accSections.concat(updatedSection)
+        }, [])
+    }
+
+    getReadme({ sectionName, componentName }) {
+        const readme = 'README.md'
+        const readmePath = path.join(filePath.examples, sectionName, componentName, readme)
+        let content = ''
+
+        try {
+            content = this.readFile(readmePath)
+        } catch (err) {
+            console.log(chalk.red(`No README provided for ${componentName}.`))
+        }
+
+        return content
+    }
+
+    getExampleData({ sectionName, componentName }) {
+        const examplesPath = path.join(filePath.examples, sectionName, componentName)
+        const examples = this.getExampleFiles({ sectionName, componentName })
+
+        return examples.map(file => {
+            const filePath = path.join(examplesPath, file)
+            const content = this.readFile(filePath)
+            let description = ''
+
+            try {
+                description = parse(content).description
+            } catch (err) {
+                console.log(chalk.red(`Could not parse description in ${filePath} due to ${err}`))
+            }
+
+            return {
+                // Remove the .js extension to get the component name.
+                name: file.slice(0, -3),
+                description,
+                code: content
+            }
+        })
+    }
+
+    getExampleFiles({ sectionName, componentName }) {
+        const examplesPath = path.join(filePath.examples, sectionName, componentName)
+        let exampleFiles = []
+        try {
+            // get all example files and filter out the README
+            exampleFiles = this.getFiles(examplesPath).filter(file => file !== 'README.md')
+            // throw error if no example files exist
+            if (exampleFiles.length === 0) {
+                throw Error
+            }
+        } catch (error) {
+            console.log(chalk.red(`No examples found for ${componentName}.`))
+        }
+        return exampleFiles
+    }
+
+    writeResult(data) {
+        if (this.errors.length) {
+            console.log(chalk.red(this.errors.join('\n')))
+        } else {
+            this.writeFile(
+                path.join(quarkPaths.docs, 'componentData.js'),
+                `module.exports = ${JSON.stringify(this.data, null, '')}`
```

Comment:
Since we'll be importing this json file into a web app bundled by webpack or parcel, we can probably do ` export default ${this.data}`